### ISSUE TITLE
Artifact actions are outdated

### ISSUE THREAD

**Issue by mbtaylor** [ivoa-std/ivoatex#140]
If I understand what's going on here correctly (and I might have got this all wrong, since I'm no expert on github CI)...

The CI actions created in `.github/workflows` by the `make github-preview` target generate actions that now fail because they use outdated versions of the github artifact action API.

For example [DALI PR#43](https://github.com/ivoa-std/DALI/pull/43) is (at time of writing) blocked because of [this report](https://github.com/ivoa-std/DALI/actions/runs/11914934507/job/33204022846?pr=43), which is complaining about actions like 
```
uses: actions/checkout@v1
```
in the context of [this github CI deprecation notice](https://github.blog/changelog/2024-02-13-deprecation-notice-v1-and-v2-of-the-artifact-actions/).

So the `github-preview` target needs updating, and it would be very helpful to have some migration instructions for workflow files in existing repositories that have been created by earlier use of this target.  It may be as simple as `s/@v[12]/@v4/`, but I'm scared to try it out.

**Comment by gmantele** [comment]
Indeed, some persons have already asked to me about few version changes. I am preparing a PullRequest connected to this Issue.

I will also try to include in this PR a fix for the releases issue: since a new version of a GitHub action, each time a Pre-Release with a new preview is published, the former Pre-Release is transformed into a Draft Release and stays there forever ; a lot of manual cleanup is needed to get rid of all of these.

**Comment by gmantele** [comment]
See the PullRequest #141 for a solution to this issue.

I created the two Makefile targets mentioned here:
- `clear-ci`: just remove the two GitHub workflow files
- `update-ci`: remove the two existing GitHub workflow files and replace them with the new version

**Comment by msdemlei** [comment]
@mbtaylor: can you try master and close this bug if it works now?  Thanks!

**Comment by mbtaylor** [comment]
No, a `make update-ci` still reports errors, see https://github.com/ivoa-std/DALI/pull/45.  It's possible that the problem does not indicate an error in #141, but I'll leave @gmantele to investigate that.

**Comment by gmantele** [comment]
So, I guess, we have to revert the installation instruction for `pdftk`. I thought it was OK now to install it using the APT package, because I tested it with my ADQL repository. But some reason, my repository don't need `pdftk` and that's why I did not get the error you have.

So, let's do that: install pdftk with snap as we did before. I submit a new PullRequest for this.

Anyway, sorry for this problem.
